### DISCUSSION TITLE
Will Cawbird be able to use Twitter's API in the future?

### DISCUSSION THREAD

**Discussion by alizehctl** [CodedOre/NewCaw#67] [General]
Twitter has [officially banned third-party clients](https://www.theverge.com/2023/1/19/23562947/twitter-third-party-client-tweetbot-twitterific-ban-rules) from using its API. Cawbird isn't in the list of clients that have been banned, but it could easily be banned in the future.

Should Twitter support be dropped in advance?

**Comment by IBBoard** [comment]
It already looked likely that there was a third-party clients ban. But I think it's fairly clear now:

> The [new rules](https://developer.twitter.com/en/developer-terms/) state that you can’t use Twitter’s API or content to “create or attempt to create a substitute or similar service or product to the Twitter Applications.”

That section on applications is new. I thought there was a generic "no integrating content into competing networks, or adding commentary on top", but I can't find it in the old terms.

As far as I can tell, the only reason Cawbird isn't banned is because we're not big enough for them to notice. They're only half way down the "list of API keys by active users" and are wading through the mass of spam systems and website integrations that have more users than us.

**Reply by IBBoard** [reply]
Update: Our keys got revoked.
